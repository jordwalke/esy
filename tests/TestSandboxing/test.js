const {exec: execBase, readFile} = require('../harness');

function exec(cmd) {
  return execBase(cmd, {cwd: __dirname});
}

function execWithFailure(cmd) {
  return execBase(cmd, {cwd: __dirname, expectFailure: true});
}

test('env', () => {
  let res = exec('../../.bin/esy');
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});

test('catches violation', () => {
  let res = execWithFailure(`
  rm -rf _build _install _esy_store
  ../../.bin/esy build`
  );
  expect(res.status).not.toBe(0);
  expect(readFile(__dirname, 'should_be_GOOD')).toBe('GOOD');
});
