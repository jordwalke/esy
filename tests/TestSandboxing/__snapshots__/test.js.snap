exports[`test env 1`] = `
"
# EsySandboxVariables /Users/andreypopp/Workspace/esy/tests/TestSandboxing
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"darwin\"
export esy__architecture=\"x64\"
export esy__target_platform=\"darwin\"
export esy__target_architecture=\"x64\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"TestSandboxing\"
export cur__version=\"1.0.0\"
export cur__source_root=\"$esy__sandbox\"
export cur__root=\"$esy__sandbox\"
export cur__depends=\"\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy_findlib.conf\"

# TestSandboxing@1.0.0 $esy__sandbox/package.json
# [BuiltIns]
export testsandboxing__name=\"TestSandboxing\"
export testsandboxing__version=\"1.0.0\"
export testsandboxing__source_root=\"$esy__sandbox\"
export testsandboxing__root=\"$esy__sandbox\"
export testsandboxing__depends=\"\"
export testsandboxing__target_dir=\"$esy__sandbox/_build\"
export testsandboxing__install=\"$esy__sandbox/_install\"
export testsandboxing__bin=\"$testsandboxing__install/bin\"
export testsandboxing__sbin=\"$testsandboxing__install/sbin\"
export testsandboxing__lib=\"$testsandboxing__install/lib\"
export testsandboxing__man=\"$testsandboxing__install/man\"
export testsandboxing__doc=\"$testsandboxing__install/doc\"
export testsandboxing__stublibs=\"$testsandboxing__install/stublibs\"
export testsandboxing__toplevel=\"$testsandboxing__install/toplevel\"
export testsandboxing__share=\"$testsandboxing__install/share\"
export testsandboxing__etc=\"$testsandboxing__install/etc\"
"
`;
